medusa/providers/torrent/html/privatehd.py

```diff
@@ -0,0 +1,184 @@
+
+        with BS4Parser(data, 'html5lib') as html:
+            if not html:
+                log.debug('No html data parsed from provider')
+                return items
+
+            torrents = html('tr')
+            if not torrents or len(torrents) < 2:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            # Skip column headers
+            for row in torrents[1:]:
+                # Skip extraneous rows at the end
+                if len(row.contents) < 10:
+                    continue
+
+                try:
+                    title = row.find(class_='torrent-filename').text.strip()
+                    download_url = row.find(class_='torrent-download-icon').get('href')
+                    size = convert_size(row.contents[11].text.strip(), -1)
+                    seeders = row.contents[13].text
+                    leechers = row.contents[15].text
+                    pubdate = self.parse_pubdate(row.contents[7].findChild().get('title'))
+
+                    item = {
+                        'title': title,
+                        'link': download_url,
+                        'size': size,
+                        'seeders': seeders,
+                        'leechers': leechers,
+                        'pubdate': pubdate,
+                    }
+                    if mode != 'RSS':
+                        pass
+                    log.debug('Found result: {0} with {1} seeders and {2} leechers',
+                              title, seeders, leechers)
+
+                    items.append(item)
+                except (AttributeError, TypeError, KeyError, ValueError, IndexError):
+                    log.error('Failed parsing provider. Traceback: {0!r}',
+                              traceback.format_exc())
+
+        return items
+
+    def login(self):
+        """Login method used for logging in before doing search and torrent downloads."""
+        if any(dict_from_cookiejar(self.session.cookies).values()):
+            return True
+
+        if 'pass' in dict_from_cookiejar(self.session.cookies):
+            return True
+
+        with BS4Parser(self.session.get(self.urls['login']).text, 'html5lib') as html:
+            token = html.find('input', attrs={'name': '_token'}).get('value')
+
+        login_params = {
+            '_token': token,
+            'email_username': self.username,
+            'password': self.password,
+            'remember': 1,
+            'submit': 'Login',
+        }
+
+        response = self.session.post(self.urls['login'], data=login_params)
+        if not response or not response.text:
+            log.warning('Unable to connect to provider')
+            return False
+
+        if 'These credentials do not match our records.' in response.text:
+            log.warning('Invalid username or password. Check your settings')
+            return False
+
+        return True
+
+    def _check_auth(self):
```

Comment:
This is not used?

Reply:
`_check_auth` is used by the `GenericProvider` class in the [`find_search_results` function](https://github.com/pymedusa/Medusa/blob/bd0832551d836713481dd0a93f4118372bb5d963/medusa/providers/generic_provider.py#L207), the provider will work without it because the [default implementation](https://github.com/pymedusa/Medusa/blob/bd0832551d836713481dd0a93f4118372bb5d963/medusa/providers/generic_provider.py#L549) always returns `True`.

Reply:
Yep, that's because most of the time it's not really needed / redundant. If you want to add it, can you please raise AUtheException like here? https://github.com/pymedusa/Medusa/blob/1a8dc66c7ec9231be0d3444710e476437c9ae1c8/medusa/providers/torrent/html/gftracker.py#L220-L221